python/ray/serve/tests/test_standalone3.py

```diff
@@ -506,5 +506,56 @@ def _check():
     serve.shutdown()
 
 
+@pytest.mark.parametrize("wait_for_controller_shutdown", (True, False))
+def test_controller_shutdown_gracefully(
```

Comment:
once you add a timeout to the ray.get for the shutdown ref, please add a test here that verifies if shutdown hangs (e.g., `graceful_shutdown_timeout_s` is super high for a deployment), the shutdown command exits after the timeout with a warning message